econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -997,12 +1023,21 @@ def _tree(self, is_policy, Xtest, feature_index, *, treatment_cost=0,
         if is_policy:
             intrp.interpret(result.estimator, Xtest,
                             sample_treatment_costs=treatment_cost)
-            treat = intrp.treat(Xtest)
-        else:  # no treatment cost for CATE trees
+            if result.feature_baseline is None:  # continuous treatment, so apply a treatment level 10% of typical
+                treatment_level = result.treatment_value * 0.1
+            else:
+                treatment_level = 1
+            # TODO: this calculation is correct only if treatment consts are marginal costs,
+            #       because then scaling the difference between treatment value and treatment costs is the
+            #       same as scaling the treatment value and subtracting the scaled treatment cost.
+            #       If treatment costs are absolute, then we'll need to modify the policy tree code itself to handle
+            #       treatment levels other than 1.
+            policy_values = intrp.policy_value_ * treatment_level, intrp.always_treat_value_ * treatment_level
```

Comment:
Here seems still unclear to me for both continue and discrete case.

```
        all_y = np.hstack([np.zeros((y_pred.shape[0], 1)), np.atleast_1d(y_pred)])

        self.tree_model_.fit(X, all_y)
        self.policy_value_ = np.mean(np.max(self.tree_model_.predict_value(X), axis=1))
        self.always_treat_value_ = np.mean(y_pred, axis=0)
``` 
The code above is how we get `policy_value_`, it's the average of our best treatment policy, where some subgroup we treat, some subgroup we don't treat, why we give a treatment level for all? should we only give a treatment level for treated sub group?

I might misunderstand it...please let me know how to explain it.

Reply:
What does this function return when treatment is categorical with many categories? Has this been tested and has it been documented in the docstring what is the form of the output?

Reply:
We should also document in the docstring that for continuous features, the treatment_cost is the "per unit treatment cost, or marginal treatment cost", i.e. that each extra unit of treatment costs that, i.e. the total cost is "treatment_cost * treatment_level". 

Does the docstring also document what values 'treatment_cost" can take when treatment is categorical with multiple categories? Can I pass a list which contains the cost for each non-baseline treatment?

Reply:
Also for categorical features, the "always treat" should be a list containing the values of always treatment with each separate non baseline treatment. Shall we then maybe also be returning the associated "names of those non-baseline treatment". So that then the UX can say: gains over always treatment wit T="A", gains over always treatment wit T="B" etc. for each bar in the bar plot

Reply:
The docstring has been updated and tests added; I have not yet added a return for the feature names.

Reply:
I've also expanded the explanation of what we compute here and why the math is correct.